### DISCUSSION TITLE
step 3 - input data

### DISCUSSION THREAD

**Discussion by colettembrown** [NGEET/fates-tutorial#37] [Testing]
In Step 3 of the Tutorial Setup, it asks that we include a local directory name in the "INPUT_DATA" field. For the workshop people will have their own data (for people new to terminal they might not know what it means to add a local directory, but I can recommend a change later), but for testing, what is the file we should use for INPUT_DATA?

**Comment by glemieux** [comment]
@colettembrown for testing you can use any directory you would like.  For simplicity, I would use the full path to the `fates-tutorial/inventor_data` directory that you downloaded/cloned to your local drive.  The `docker compose up -d` command will update this directory with a `bci` subfolder and files associated with the Barro Colorado Island site.  I'll make sure to clarify that in the tutorial setup.

**Reply by colettembrown** [reply]
@glemieux thanks. I was already working in fates-tutorial so set it to "/inventory_data" and I saw the initial image in step 4.3. in [the tutorial](https://fates-users-guide.readthedocs.io/projects/tutorial/en/latest/docker_introduction.html). However, it eventually ended with "Error response from daemon: Mounts denied: 
The path /inventory_data is not shared from the host and is not known to Docker.
You can configure shared paths from Docker -> Preferences... -> Resources -> File Sharing."

I set up file sharing in my docker desktop as per their instructions. Then, I reran "docker compose up -d" in terminal and expected to see the same output from the tutorial docs (step 4.3. output), but I see the attached image instead where "Container tutorial-notebook" is "Starting" instead of "Started" and I get the same error in terminal. 
![image](https://github.com/user-attachments/assets/a3f2975c-29b8-4370-93a9-dbbab8491636)

**Reply by glemieux** [reply]
Hmmm that's a new one to me :thinking:, good question.  Two follow up questions:

1.  In the `.env` file are you using the full path for `INPUT_DATA` to the inventory file or are you using the relative path?  I think it might need to be the full path.
2. In the docker desktop settings pane, is your home directory listed under the "Resources/File sharing" tab?

Oh, also what OS are you using?

**Reply by colettembrown** [reply]
1. No it was not, I updated it to be my full path. Sorry I missed that in your earlier instructions. It had to be "/Users/colettebrown/repos/fates-tutorial/inventory_data" and I had to add that entire path to my "Resources/File Sharing" in Docker settings.
2. When you say home directory, do you mean home directory for the fates-tutorial? This image is all of the directories in the File sharing tab. 
<img width="736" alt="image" src="https://github.com/user-attachments/assets/28043cb8-0ca9-4bd4-97c1-46ad11cf5a99">

3. I am on a Mac M3 - Sonoma 14.3

When I rerun "docker compose up -d" I no longer get the error message above, I get a new error (image below) with the error message: The requested image's platform (linux/amd64) does not match the detected host platform (linux/arm64/v8) and no specific platform was requested 0.0s 
![image](https://github.com/user-attachments/assets/16830b78-f837-4430-a6e8-89c4147fc660)

**Reply by glemieux** [reply]
Copy that.  I'll need a little more time to track this new one down.

To answer your reply to my second question, I meant the home directory for your user account on your laptop.  Sorry for the confusion.  Your sharing options look correct.

**Reply by glemieux** [reply]
@colettembrown the arm64 images have been rebuilt and updated.  Would you delete all the existing images and volumes from your machine and re-run the `docker compose up -d` command, please?

**Comment by LEBoeschoten** [comment]
@glemieux Hi Gregory! I have a problem at the same step. I've updated the .env with the full path, but when trying to start the containers it gives me the following error message. Running on Windows. Any thoughts? Thank you!

![image](https://github.com/user-attachments/assets/8040fe5b-42ab-459d-9d66-853adb0615b5)

**Reply by glemieux** [reply]
Hi @LEBoeschoten, I realized this is due to an oversight on my part!  I will send out an email update today with the fix and reply here as well.  Apologies for the confusion.

**Reply by LEBoeschoten** [reply]
All good - thank you!

**Reply by LEBoeschoten** [reply]
I just downloaded the repo again and logged in with my work email address in Docker (I'd logged in with my github before), which fixed it! It's running now.

**Reply by LEBoeschoten** [reply]
Plus I restored the docker settings to factory defaults by clicking the '?' icon in the top > troubleshooting >reset to factory defaults, and restarted my computer.
